diff --git a/packages/client/src/scripts/get-note-summary.ts b/packages/client/src/scripts/get-note-summary.ts
index bd394279c..6cd792b89 100644
--- a/packages/client/src/scripts/get-note-summary.ts
+++ b/packages/client/src/scripts/get-note-summary.ts
@@ -33,6 +33,7 @@ export const getNoteSummary = (note: misskey.entities.Note): string => {
 		summary += ` (${i18n.locale.poll})`;
 	}
 
+	/*
 	// 返信のとき
 	if (note.replyId) {
 		if (note.reply) {
@@ -50,6 +51,7 @@ export const getNoteSummary = (note: misskey.entities.Note): string => {
 			summary += '\n\nRN: ...';
 		}
 	}
+	*/
 
 	return summary.trim();
 };
