diff --git a/packages/backend/src/server/activitypub.ts b/packages/backend/src/server/activitypub.ts
index cd5f917c4..c05999ce7 100644
--- a/packages/backend/src/server/activitypub.ts
+++ b/packages/backend/src/server/activitypub.ts
@@ -41,17 +41,20 @@ function inbox(ctx: Router.RouterContext) {
 }
 
 const ACTIVITY_JSON = 'application/activity+json; charset=utf-8';
+const JSON = 'application/json; charset=utf-8';
 const LD_JSON = 'application/ld+json; profile="https://www.w3.org/ns/activitystreams"; charset=utf-8';
 
 function isActivityPubReq(ctx: Router.RouterContext) {
 	ctx.response.vary('Accept');
-	const accepted = ctx.accepts('html', ACTIVITY_JSON, LD_JSON);
+	const accepted = ctx.accepts('html', ACTIVITY_JSON, JSON, LD_JSON);
 	return typeof accepted === 'string' && !accepted.match(/html/);
 }
 
 export function setResponseType(ctx: Router.RouterContext) {
-	const accept = ctx.accepts(ACTIVITY_JSON, LD_JSON);
-	if (accept === LD_JSON) {
+	const accept = ctx.accepts(ACTIVITY_JSON, JSON, LD_JSON);
+	if (accept === JSON) {
+		ctx.response.type = JSON;
+	} else if (accept === LD_JSON) {
 		ctx.response.type = LD_JSON;
 	} else {
 		ctx.response.type = ACTIVITY_JSON;
