diff --git a/src/client/components/google.vue b/src/client/components/google.vue
index be724f038..aa8e5e586 100644
--- a/src/client/components/google.vue
+++ b/src/client/components/google.vue
@@ -26,7 +26,7 @@ export default defineComponent({
 	},
 	methods: {
 		search() {
-			window.open(`https://www.google.com/search?q=${this.query}`, '_blank');
+			window.open(`https://lite.qwant.com/?q=${this.query}`, '_blank');
 		}
 	}
 });
diff --git a/src/mfm/to-html.ts b/src/mfm/to-html.ts
index b3678a0dd..3e755b24f 100644
--- a/src/mfm/to-html.ts
+++ b/src/mfm/to-html.ts
@@ -147,7 +147,7 @@ export function toHtml(nodes: mfm.MfmNode[] | null, mentionedRemoteUsers: IMenti
 
 		search(node) {
 			const a = doc.createElement('a');
-			a.href = `https://www.google.com/search?q=${node.props.query}`;
+			a.href = `https://lite.qwant.com/?q=${node.props.query}`;
 			a.textContent = node.props.content;
 			return a;
 		}
