diff --git a/src/client/components/google.vue b/src/client/components/google.vue
index 6d8ae0b5b..c71d20307 100644
--- a/src/client/components/google.vue
+++ b/src/client/components/google.vue
@@ -22,7 +22,7 @@ export default defineComponent({
 	methods: {
 		search() {
 			const engine = this.$store.state.webSearchEngine ||
-				'https://www.google.com/search?q={{query}}';
+				'https://metager.de/meta/meta.ger3?eingabe={{query}}';
 			const url = engine.replace('{{query}}', this.query)
 			window.open(url, '_blank');
 		}
diff --git a/src/mfm/to-html.ts b/src/mfm/to-html.ts
index aa39443c6..6dd4852f5 100644
--- a/src/mfm/to-html.ts
+++ b/src/mfm/to-html.ts
@@ -147,7 +147,7 @@ export function toHtml(nodes: mfm.MfmNode[] | null, mentionedRemoteUsers: IMenti
 
 		search(node) {
 			const a = doc.createElement('a');
-			a.href = `https://www.google.com/search?q=${node.props.query}`;
+			a.href = `https://metager.de/meta/meta.ger3?eingabe=${node.props.query}`;
 			a.textContent = node.props.content;
 			return a;
 		}
